src/ortho.rs

```diff
@@ -93,6 +93,35 @@ impl Ortho {
         }
         idx
     }
+    pub fn subtract(&self) -> Option<Self> {
+        // Find the last filled position (position of last Some value before first None)
+        let current_position = self.get_current_position();
+        if current_position == 0 {
+            // Nothing to subtract - ortho is empty
+            return None;
+        }
+        
+        let last_filled_position = current_position - 1;
+        
+        // Create new payload with the last filled value removed
+        let mut new_payload = self.payload.clone();
+        new_payload[last_filled_position] = None;
+        
+        // Note: We don't reverse canonicalization here. The canonicalization that
```

Comment:
There is a mistake in subtract. There are two places in add where payload is rearranged. It is rearranged in the special case of potentially swapping two and three, and for that example it is safe to simply remove. There Is a more advanced case. The general case calls into spatial for a rearrange pattern. This has to be done in reverse for subtract to work. The other issue is that subtract may change shape. Make sure shape comes out right.  Last issue is that subtract may lead to duplicates in the new work queue. Deduplicate those by ID.